lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+/* global document,window */
+
+/* istanbul ignore next */
+function collectTagsThatBlockFirstPaint() {
```

Comment:
yeah, I mean a lot of this can be done with `driver.querySelectorAll` and then dealing with the `Element`s returned. Biggest issue might be `window.matchMedia()`, but there might be some existing way to do that with the devtools protocol (short of `evaluateAsync`ing just that).

It certainly does make debugging a lot easier when more is done outside the context of the page, but I don't have a strong reaction against this compared to the previous `<link>` only version